docs/modules/ROOT/pages/reference/architecture.adoc

```diff
@@ -0,0 +1,170 @@
+= Architecture
+
+Commodore's architecture roughly corresponds to the diagram below.
+
+// TODO architecture diagram
```

Comment:
Might need to be done or removed before this can be merged.

Reply:
Removed for now since I don't even remember what I envisioned here.